### DISCUSSION TITLE
Renovate fails to check Swift Package Manager updates when tag starts with `v`

### DISCUSSION THREAD

**Discussion by bdolewski-intellias** [renovatebot/renovate#28810] [Request Help]
### What would you like help with?

I think I found a bug

### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

GitHub

### Please tell us more about your question or problem

We started using Renovate to maintain both GitHub Actions dependencies and Swift Package Manager updates.

Renovate rocks 🚀! But we found a problem with some of our internal repositories.

When SPM package (repo) has a release with git tag that starts with `v`-prefix (for example `v6.0.0`) then it fails:
```json
{
      "datasource": "git-tags",
      "depName": "my-company/MyPackage-Swift",
      "packageName": "https://github.com/my-company/MyPackage-Swift",
      "currentValue": "6.0.0",
      "updates": [],
      "versioning": "swift",
      "warnings": [],
      "sourceUrl": "https://github.com/my-company/MyPackage-Swift",
      "skipReason": "invalid-value"
    }
```
<img width="585" alt="Screenshot 2024-05-03 at 07 43 10" src="https://github.com/renovatebot/renovate/assets/109529565/d1947981-6b79-4c6a-a877-9450cd6e57a5">

However Renovate works perfectly fine SPM packages that drop `v`:
https://github.com/pointfreeco/swift-composable-architecture/tags

SPM access both styles, Dependabot is happy with that either (because it uses Swift CLI tools under the hood)

I think that Renovate for SPM should allow tags to start with `v` - since it's okay for Swift Package Manager itself.

### Logs (if relevant)

<details><summary>Logs</summary>

```

Replace this text with your logs, between the starting and ending triple backticks

```

</details>

**Comment by rarkins** [comment]
What does the package file in your repo look like? Can you reproduce this with a public example?

**Reply by bdolewski-intellias** [reply]
@rarkins 
> What does the package file in your repo look like?

Everything looks like this in my repo:
```swift
// Internal
.package(url: "https://github.com/my-org/Package1-Swift", exact: "5.3.0"),
.package(url: "https://github.com/my-org/Package2-Swift", exact: "6.1.0"),

// External Dependencies: PointFree
.package(url: "https://github.com/pointfreeco/swift-composable-architecture", exact: "1.7.3"),
.package(url: "https://github.com/pointfreeco/swift-dependencies", exact: "1.2.2"),
.package(url: "https://github.com/pointfreeco/swift-snapshot-testing", exact: "1.16.0"),
.package(url: "https://github.com/pointfreeco/swiftui-navigation", exact: "1.3.0"),
```

Initially I thought that this issue is triggered by `.git` at the end of package's URL, like this:
```swift
.package(url: "https://github.com/pointfreeco/swift-snapshot-testing.git", exact: "1.16.0"),
```

because one of few internal packages that were picked by Renovate had `.git` at the end. After all it was simply coincidence because these packages had no `v`. Good riddance that other developers forgot to follow this "convention" for SPM to add `v` 😊

**Conclusion**

`.git` doesn't change anything

**Reply by bdolewski-intellias** [reply]
@rarkins 
> Can you reproduce this with a public example?

Yeah, I think I can do that. I'll need 1 demo repo, 1 demo dependency repo with tag `v1.0.0` and one more dependency without `v`.

Do I need to add Renovate to demo project?

**Reply by rarkins** [reply]
It's hopefully a quick fix as soon as I can understand where exactly the problem is. A reproduction is usually the best way for that

**Reply by bdolewski-intellias** [reply]
@rarkins 
Here you go: https://github.com/bdolewski-intellias/renovate-spm-demo

Verified in Xcode that both dependencies are ok (with and without `v`)
<img width="1512" alt="Screenshot 2024-05-03 at 08 46 48" src="https://github.com/renovatebot/renovate/assets/109529565/252e0d6e-c90b-4ccf-8cf3-9c3ddedb6e42">
<img width="773" alt="Screenshot 2024-05-03 at 08 47 36" src="https://github.com/renovatebot/renovate/assets/109529565/9e7d6add-8426-4145-8ae6-4943f6f78ac9">

**Reply by rarkins** [reply]
Can you make one with and one without the v prefix?

**Reply by bdolewski-intellias** [reply]
> Can you make one with and one without the v prefix?

This one is without `v`: `.package(url: "https://github.com/pointfreeco/swift-composable-architecture", exact: "1.7.3"),`
This one has `v`: `.package(url: "https://github.com/bdolewski-intellias/renovate-spm-dependency", exact: "1.0.0"),`

**Reply by rarkins** [reply]
Can you create a valid update for it (v1.0.1) so I can also test that it works?

**Reply by bdolewski-intellias** [reply]
@rarkins done!

**Comment by rarkins** [comment]
Fix PR: https://github.com/renovatebot/renovate/pull/28822

**Reply by bdolewski-intellias** [reply]
@rarkins you made my day! Thanks a lot and!
